jakarta-eclipse/src/org/jakarta/codeAction/HttpServletQuickFix.java

```diff
@@ -0,0 +1,42 @@
+package org.jakarta.codeAction;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.ITypeBinding;
+import org.eclipse.jdt.internal.corext.dom.Bindings;
+import org.eclipse.jdt.internal.core.manipulation.util.BasicElementLabels;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+
+import org.jakarta.jdt.ServletConstants;
+import org.jakarta.codeActionProposal.ChangeCorrectionProposal;
+
+public class HttpServletQuickFix implements IJavaCodeActionParticipant{
+	@Override
+	public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
+			IProgressMonitor monitor) throws CoreException {
+		ASTNode node = context.getCoveredNode();
+		ITypeBinding parentType = Bindings.getBindingOfParentType(node);
+		if (parentType != null) {
+			List<CodeAction> codeActions = new ArrayList<>();
+			// Create code action
+			// interface
+			final String TITLE_MESSAGE = "Let ''{0}'' implement ''{1}''";
+			String args[] = { BasicElementLabels.getJavaElementName(parentType.getName()),
+					BasicElementLabels.getJavaElementName(ServletConstants.HTTP_SERVLET) };
+			ChangeCorrectionProposal proposal = new ChangeCorrectionProposal(parentType.getName(),
+					ServletConstants.HTTP_SERVLET, );
```

Comment:
The example was using the comment out line below to transfer an implementationProposal into a changeCorrectionProposal. 

changeCorrectionProposal takes three params (name, kind, change)

As they have different constructors, and implementationProposal does not use any change class or object.

I am not sure how it sets the param change for ChangeCorrectionProposal. Please give me some insights, thank you!

Reply:
@kathrynkodama @rzgry plz take a look. Thank you

Reply:
I wouldn't try to focus on the `ImplementationProposal` and just focus on the `ChangeCorrectionProposal`. If you need to change the constructors that's fine. If it makes more sense to just use those three params(name, kind, change) that is fine. Let me know if that answers your question.

Reply:
ok, I will only use the `ChangeCorrectionProposal ` 

As for my question, I am more of wondering how to initiate the change param inside the proposal class.  Because there is no relative definition around.

Reply:
Hm, I'm not sure if I understand your question - let's discuss on slack